File: lista 15questoes/15 questoes em python/att04.py
#4. Herança Crie uma classe base Animal com métodos como som. Derive classes como Cachorro e Gato que implementam o método som de maneiras diferentes. 

class Animal:
    def __init__(self,nome):
        self.nome = nome

    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        print(f"{self.nome} faz: Au Au")

class Gato(Animal):
    def som(self):
        print(f"{self.nome} faz: miau miau")

cachorro1 = Cachorro("scooby")
gato1 = Gato("Mu")

cachorro1.som()
gato1.som()

File: lista 15questoes/15 questoes em java/att04.java
class Animal {
    String nome;

    Animal(String nome) {
        this.nome = nome;
    }

    void som() {}
}

class Cachorro extends Animal {
    Cachorro(String nome) {
        super(nome);
    }

    void som() {
        System.out.println(nome + " faz: Au Au");
    }
}

class Gato extends Animal {
    Gato(String nome) {
        super(nome);
    }

    void som() {
        System.out.println(nome + " faz: Miau Miau");
    }
}

class Main {
    public static void main(String[] args) {
        Cachorro cachorro1 = new Cachorro("Scooby");
        Gato gato1 = new Gato("Mu");
        cachorro1.som();
        gato1.som();
    }
}